### PR TITLE
topic explorer

### PR DESCRIPTION
**Pull request by belljohan3**

### CODE REVIEW

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+const pinnedTopics = useDexieLiveQueryWithDeps(
+    appLanguageIdAsRef,
+    (appLanguageId: Uuid) =>
+        db.docs
+            .where({
+                type: DocType.Content,
+                language: appLanguageId,
+                status: "published",
+                parentPinned: 1, // 1 = true
+                // parentTagType: TagType.Topic,
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:25]: Commented line. Should it perhaps be un-commented?

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                if (!content.publishDate) return false;
+                if (content.publishDate > Date.now()) return false;
+                if (content.expiryDate && content.expiryDate < Date.now()) return false;
+
+                for (const tagId of content.parentTags) {
+                    if (pinnedTopics.some((p) => p.parentId == tagId)) return true;
+                }
+                return false;
+            })
+            // .orderBy("updatedTimeUtc") as unknown as Promise<ContentDto[]>,
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:62]: Commented line

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+const pinnedTopicContent = useDexieLiveQueryWithDeps(
+    [appLanguageIdAsRef, pinnedTopics],
+    ([appLanguageId, pinnedTopics]: [Uuid, ContentDto[]]) =>
+        db.docs
+            .where({
+                type: DocType.Content,
+                language: appLanguageId,
+                status: "published",
+            })
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:50]: This is a very heavy query, which can be improved when #463 is implemented. Please create a separate ticket to redo this query?

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                }
+                return false;
+            })
+            // .orderBy("updatedTimeUtc") as unknown as Promise<ContentDto[]>,
+            .toArray() as unknown as Promise<ContentDto[]>,
+    { initialValue: await db.getQueryCache<ContentDto[]>("explore_pinnedContent") },
+);
+
+watch(pinnedTopicContent, async (value) => {
+    db.setQueryCache<ContentDto[]>("explorepage_pinnedContent", value);
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:68]: The cache key is not matching the key used in the query

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+});
+
+// sort pinned content by category
+const pinnedContentByTopic = contentByTopic(pinnedTopicContent, pinnedTopics);
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:72]: contentByTopic can be replaced with contentByTag when #463 is implemented

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+</script>
+
+<template>
+    <div class="-my-6">
+        <IgnorePagePadding>
+            <template v-for="c in pinnedContentByTopic" :key="c.topic._id">
+                <!-- Check if the filtered content has any items -->
+                <HorizontalContentTileCollection
+                    v-if="
+                        c.content.filter((content: ContentDto) => content.parentTagType === 'topic')
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:82]: Do we need this filter?

File: app/src/components/ExplorePage/ExplorePinnedTopics.vue [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                            .length > 0
+                    "
+                    :contentDocs="
+                        c.content.filter((content: ContentDto) => content.parentTagType === 'topic')
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/ExplorePinnedTopics.vue:86]: Do we need this filter?

File: app/src/components/ExplorePage/contentByTopic.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
+import { ref, watchEffect, type Ref } from "vue";
```
Comment (ivanslabbert) [inline app/src/components/ExplorePage/contentByTopic.ts:1]: contentByTopic can be replaced by contentByTag when #463 is implemented

**Comment by belljohan3** [comment]
Hey @ivanslabbert ! i have rebase this to main but it is still showing that is out-of-date

**Comment by ivanslabbert** [comment]
> Hey @ivanslabbert ! i have rebase this to main but it is still showing that is out-of-date
Perhaps the main branch was outdated on your machine when you did the rebase?

**Comment by ivanslabbert** [comment]
> image Think we need to truncate the summary text.
> 
> image Here the image is scaled when the text is too long?
> 
> image It seems as if there is an additional margin on the left side.


Seems like these issues are not fixed yet?

**Comment by belljohan3** [comment]
Something is weird. I fixed the issues today.

**Comment by ivanslabbert** [comment]
- [ ] Fix pinned category query on topics page to show categories containing pinned topics (with the topic content docs listed) - similar to the home page pinned categories query
- [ ] Add missing indexes
- [ ] Implement infinite scrolling for topics list

**Review by DirkDevelopmentFreelance — DISMISSED** [review]
I am missing an option to toggle a grid view, on desktop mode having only a list view isn't a good experience in my opinion, a grid view would work better for that as a default and then on mobile view a user can have the option to toggle how they receive the data. I feel this is an important feature to have. Even though we know most of the users will be on mobile phones, some people would maybe also like to watch on their tv's after discovering the app on their phones, or on a large monitor connected to a pc.

**Comment by ivanslabbert** [comment]
> I am missing an option to toggle a grid view, on desktop mode having only a list view isn't a good experience in my opinion, a grid view would work better for that as a default and then on mobile view a user can have the option to toggle how they receive the data. I feel this is an important feature to have. Even though we know most of the users will be on mobile phones, some people would maybe also like to watch on their tv's after discovering the app on their phones, or on a large monitor connected to a pc.

This is moved to a new issue: #575

**Review by ivanslabbert — APPROVED** [review]

**Pull request merged**